### PR TITLE
grpc access logger: add downstream transport failure reason

### PR DESCRIPTION
**Pull request by kralicky**
Commit Message: grpc access logger: add downstream transport failure reason

Additional Description:
This adds the missing downstream transport failure reason field to the grpc access logger. The field is already available when using the file logger, but is not populated when using the grpc logger.

Risk Level: low
Testing: yes, added a few integration tests; some small changes made to test fixtures to accommodate

Docs Changes: 
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by kralicky** [comment]
I have a follow-up patch (https://github.com/envoyproxy/envoy/commit/1e01b13f3f4e85f806ca02a3189acd931110a5f4) to fix the duplicated `:TLS_error_end:TLS_error_end` suffix that I am currently checking for in the tests. If you want I can add it to this PR, or submit a second one after this since it's sorta unrelated.

File: test/extensions/access_loggers/grpc/tcp_grpc_access_log_integration_test.cc [hunk 1]
```diff
@@ -786,5 +795,186 @@ TEST_P(TcpGrpcAccessLogIntegrationTest, SslNotTerminatedWithJA3NoSNI) {
...
+  ssl_options.setTlsVersion(envoy::extensions::transport_sockets::tls::v3::TlsParameters::TLSv1_2);
+  setupSslConnection(/*expect_connection_open=*/false,
+                     /*ssl_options=*/ssl_options, /*curves_list=*/"P-256");
+
+  ASSERT_TRUE(waitForAccessLogConnection());
+  ASSERT_TRUE(waitForAccessLogStream());
+  ASSERT_TRUE(
+      // note: no peer certificate properties here, because the handshake fails
+      // at an earlier stage
+      waitForAccessLogRequest(fmt::format(R"EOF(
```
Comment (alyssawilk) [inline test/extensions/access_loggers/grpc/tcp_grpc_access_log_integration_test.cc:880]: I suspect this is going to be fragile with any access logs changes.  consider looking for specific strings instead

**Review by alyssawilk — COMMENTED** [review]
Code looks good.  please take a look at CI failures?
/retest
/wait

File: test/extensions/access_loggers/grpc/tcp_grpc_access_log_integration_test.cc (hunk 1 above)
Comment (kralicky) [reply test/extensions/access_loggers/grpc/tcp_grpc_access_log_integration_test.cc:880]: This is how the other tests in this file worked, though I do agree that it's a bit unsightly. The existing test fixture is only set up to match on entire messages. What would you suggest?

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**